parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesWriter.java

```diff
@@ -0,0 +1,150 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.bytestreamsplit;
+
+import org.apache.parquet.bytes.ByteBufferAllocator;
+import org.apache.parquet.bytes.BytesInput;
+import org.apache.parquet.bytes.CapacityByteArrayOutputStream;
+import org.apache.parquet.column.Encoding;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.io.ParquetEncodingException;
+
+public abstract class ByteStreamSplitValuesWriter extends ValuesWriter {
+
+  protected final int numStreams;
+  protected final int elementSizeInBytes;
+  private CapacityByteArrayOutputStream[] byteStreams;
+
+  public ByteStreamSplitValuesWriter(int elementSizeInBytes, int initialCapacity, int pageSize, ByteBufferAllocator allocator) {
+    this.numStreams = elementSizeInBytes;
+    this.elementSizeInBytes = elementSizeInBytes;
+    this.byteStreams = new CapacityByteArrayOutputStream[elementSizeInBytes];
+
+    // Round-up the capacity hint.
+    final int capacityPerStream = (pageSize + this.numStreams - 1) / this.numStreams;
+    final int initialCapacityPerStream = (initialCapacity + this.numStreams - 1) / this.numStreams;
+    for (int i = 0; i < this.numStreams; ++i) {
+      this.byteStreams[i] = new CapacityByteArrayOutputStream(
+              initialCapacityPerStream, capacityPerStream, allocator);
+    }
+  }
+
+  @Override
+  public long getBufferedSize() {
+    long totalSize = 0;
+    for (CapacityByteArrayOutputStream stream : this.byteStreams) {
+      totalSize += stream.size();
+    }
+    return totalSize;
+  }
+
+  @Override
+  public BytesInput getBytes() {
+    BytesInput[] allInputs = new BytesInput[this.numStreams];
+    for (int i = 0; i < this.numStreams; ++i) {
+      allInputs[i] = BytesInput.from(this.byteStreams[i]);
+    }
+    return BytesInput.concat(allInputs);
+  }
+
+  @Override
+  public Encoding getEncoding() {
+    return Encoding.BYTE_STREAM_SPLIT;
+  }
+
+  @Override
+  public void reset() {
+    for (CapacityByteArrayOutputStream stream : this.byteStreams) {
+      stream.reset();
+    }
+  }
+
+  @Override
+  public void close() {
+    for (CapacityByteArrayOutputStream stream : byteStreams) {
+      stream.close();
+    }
+  }
+
+  protected void scatterBytes(byte[] bytes) {
+    if (bytes.length != this.numStreams) {
+      throw new ParquetEncodingException("Number of bytes doesn't match the number of streams.");
```

Comment:
Add the value of bytes.length  and this.numStreams into the excepton

Reply:
Done.